crates/ursa-telemetry/src/lib.rs

```diff
@@ -68,14 +69,16 @@ impl TelemetryConfig {
     }
 
     pub fn init(self) -> anyhow::Result<()> {
-        let env_filter = EnvFilter::try_from_default_env().unwrap_or_else(|_| {
-            EnvFilter::new(self.log_level.unwrap_or_else(|| "INFO".to_string()))
-        });
+        let log_level = self
+            .log_level
+            .unwrap_or_else(|| env::var("RUST_LOG").unwrap_or_else(|_| "INFO".to_string()));
 
         let mut tracing_layers = vec![];
 
         if self.pretty_log {
-            let log_subscriber = fmt::layer().pretty().with_filter(env_filter);
+            let log_subscriber = fmt::layer()
+                .pretty()
+                .with_filter(EnvFilter::new(&log_level));
```

Comment:
Similarly here, you moved instantiating here?

Reply:
yes env filter is not clonable